tools/issue-labeler/src/IssueLabelerService/Answers/RagQna.cs

```diff
@@ -0,0 +1,146 @@
+                .Select(rd => new
+                {
+                    rd.Item1.chunk,
+                    rd.Item1.Url,
+                    Score = rd.Item2
+                })
+                .ToList();
+
+            var issues = relevantIssues
+                .Where(r => r.Item2 >= scoreThreshold)
+                .Select(rd => new
+                {
+                    rd.Item1.Title,
+                    rd.Item1.chunk,
+                    rd.Item1.Service,
+                    rd.Item1.Category,
+                    rd.Item1.Url,
+                    Score = rd.Item2
+                })
+                .ToList();
+
+            // Filtered out all sources for either one then not enough information to answer the issue. 
+            if (docs.Count() == 0 || issues.Count() == 0)
+            {
+                throw new Exception($"Not enough relevant sources found for {issue.RepositoryName} using the Complete Triage model for issue #{issue.IssueNumber}.");
+            }
+
+            double highestScore = Math.Max(docs.Max(d => d.Score), issues.Max(d => d.Score));
+            bool solution = highestScore >= solutionThreshold;
+
+            _logger.LogInformation($"Highest relevance score among the sources: {highestScore}");
+
+            // Makes it nicer for the model to read (Can probably be made more readable but oh well)
+            var printableIssues = issues.Select(r => JsonConvert.SerializeObject(r)).ToList();
+            var printableDocs = docs.Select(r => JsonConvert.SerializeObject(r)).ToList();
+
+            string instructions = _config.Instructions;
+            string message;
+            if (solution)
+            {
+                message = $"Sources:\nDocumentation:\n{string.Join("\n", printableDocs)}\nGitHub Issues:\n{string.Join("\n", printableIssues)}\nThe user needs a solution to their GitHub Issue:\n{query}";
+            }
+            else
+            {
+                message = $"Sources:\nDocumentation:\n{string.Join("\n", printableDocs)}\nGitHub Issues:\n{string.Join("\n", printableIssues)}\nThe user needs suggestions for their GitHub Issue:\n{query}";
+            }
+
+            // Structured output for the model
+            var structure = BinaryData.FromBytes("""
+            {
+              "type": "object",
+              "properties": {
+                "Category": { "type": "string" },
+                "Service": { "type": "string" },
+                "Response": { "type": "string" }
+              },
+              "required": [ "Category", "Service", "Response" ],
+              "additionalProperties": false
+            }
+            """u8.ToArray());
+
+            var response = await _ragService.SendMessageQnaAsync(instructions, message, structure);
+
+            var resultObj = JsonConvert.DeserializeObject<AIOutput>(response);
+            string intro, outro;
+            Dictionary<string, string> replacements = new Dictionary<string, string>
```

Comment:
```suggestion
            var replacements = new Dictionary<string, string>
```

Everything else around here is `var`.   Weird that this one isn't.